File: ruby/lib/deck.rb
class Deck
  attr_accessor :cards

  SUITS = %w{s h d c}
  VALUES = %w{2 3 4 5 6 7 8 9 10 J Q K A}

  def initialize options={}
    @cards = []

    SUITS.each do |s|
      VALUES.each do |v|
        cards.push "#{ v }#{ s }"
      end
    end
  end

end

File: javascript/src/Deck.js
var Deck = function(){

}

Deck.prototype.cards = function(){
  var suits = ['s','d','h','c'],
      values = ['2','3','4','5','6','7','8','9','10','J','Q','K','A'],
      cards = []
  
  for(var i = suits.length; i-->0;){
    for(var n = values.length; n-->0;){
      cards.push(values[i]+values[n])
    }
  }

  return cards;
}
